tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -1,170 +1,154 @@
-  applyUIStateChanges,
-  createFakeEvent,
-  dispatchClientMetadata,
-  fakeI18n,
-  shallowUntilTarget,
-} from 'tests/unit/helpers';
+import ConfirmButton, { extractId } from 'amo/components/ConfirmButton';
+import { render as defaultRender, screen } from 'tests/unit/helpers';
 
 describe(__filename, () => {
-  const getProps = ({ i18n = fakeI18n(), ...props } = {}) => {
-    return {
-      i18n,
-      id: 'Collection-confirm-delete',
-      message: 'some warning message',
-      onConfirm: sinon.stub(),
-      store: dispatchClientMetadata().store,
-      ...props,
-    };
-  };
+  const defaultChildText = 'the default text of this button';
 
-  const render = ({ children, ...otherProps } = {}) => {
-    return shallowUntilTarget(
-      <ConfirmButton {...getProps(otherProps)}>
-        {children || 'the default text of this button'}
+  const render = ({
+    children = defaultChildText,
+    message = 'some warning message',
+    id = 'Some-confirm-button',
+    onConfirm = jest.fn(),
+    ...otherProps
+  } = {}) => {
+    return defaultRender(
+      <ConfirmButton
+        onConfirm={onConfirm}
+        message={message}
+        id={id}
+        {...otherProps}
+      >
+        {children}
       </ConfirmButton>,
-      ConfirmButtonBase,
     );
   };
 
-  const renderWithDialog = ({
-    store = dispatchClientMetadata().store,
-    ...otherProps
-  } = {}) => {
-    const root = render({ store, ...otherProps });
+  const renderWithDialog = ({ ...otherProps } = {}) => {
+    render(otherProps);
 
     // Click to open ConfirmationDialog.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
-
-    return root;
-  };
-
-  const getDialogProp = (root, propName) => {
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp(propName);
-    return confirmDialog.prop(propName);
+    userEvent.click(
+      screen.getByRole('button', {
+        name: defaultChildText,
+      }),
+    );
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
   };
 
   it('renders a button', () => {
-    const root = render();
+    render();
 
-    expect(root).toHaveClassName('ConfirmButton');
-    expect(root.find(Button)).toHaveLength(1);
-    expect(root.find(Button)).toHaveClassName('ConfirmButton-default-button');
-    expect(root.find(Button)).toHaveProp('buttonType', 'neutral');
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('passes the buttonType prop to the button', () => {
-    const buttonType = 'alert';
-    const root = render({ buttonType });
+    render({ buttonType: 'alert' });
 
-    expect(root.find(Button)).toHaveProp('buttonType', buttonType);
+    const button = screen.getByRole('button', {
```

Comment:
Because `button` is now only used once, we could inline this into the `expect()`.